hal-core/src/boot.rs

```diff
@@ -18,6 +18,10 @@ pub trait BootInfo {
         None
     }
 
+    fn subscriber(&self) -> Option<tracing::Dispatch> {
+        None
+    }
+
     // TODO(eliza): figure out a non-bad way to represent boot command lines (is
     // it reasonable to convert them to rust strs when we barely have an operating
     // system???)
```

Comment:
Completely unrelated, but I think it'd be reasonable to do something like that. We can panic early during boot if a boot command line isn't valid utf-8 or something.